webapp/index.js

```diff
@@ -0,0 +1,253 @@
+        if (!acc.files.find(el => el.file === file)) {
+          acc.files.push({
+            file: file,
+            data: [
+              [
+                new Date(element.StartMinute).getTime(),
+                element.Nodes.all.Files[file]
+              ]
+            ]
+          });
+        } else {
+          acc.files.forEach(el => {
+            if (el.file === file) {
+              el.data.push([
+                new Date(element.StartMinute).getTime(),
+                element.Nodes.all.Files[file]
+              ]);
+            }
+          });
+        }
+      });
+      Object.keys(element.Nodes).forEach(node => {
+        if (!acc.nodes.find(el => el.node === node)) {
+          acc.nodes.push({
+            node: node,
+            data: [
+              [
+                new Date(element.StartMinute).getTime(),
+                element.Nodes[node].Volume
+              ]
+            ]
+          });
+        } else {
+          acc.nodes.forEach(el => {
+            if (el.node === node) {
+              el.data.push([
+                new Date(element.StartMinute).getTime(),
+                element.Nodes[node].Volume
+              ]);
+            }
+          });
+        }
+      });
+
+      return acc;
+    },
+    { files: [{ file: "all", data: [] }], nodes: [] }
+  );
+  const sortedFilesData = filesData.files.sort((a, b) => {
+    return (
+      b.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0) -
+      a.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0)
+    );
+  });
+
+  // choose only top 10 files or nodes and total qty
+  const sortedNodesData = filesData.nodes.sort((a, b) => {
+    return (
+      b.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0) -
+      a.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0)
+    );
+  });
+  return {
+    files: sortedFilesData.slice(0, 11),
+    nodes: sortedNodesData.slice(0, 11)
+  };
+}
+
+// date and time format for title and tooltips
+const dateTimeOptions = {
+  year: "numeric",
+  month: "short",
+  day: "2-digit",
+  hour: "2-digit",
+  minute: "2-digit"
+};
+
+// default charts are for last 10 days with 144 minutes aggregate
+async function loadAndDraw({
+  from = new Date(new Date().setDate(new Date().getDate() - 10)),
+  to = new Date(),
+  aggregate = 144
+} = {}) {
+  const jsonData = await loadData({ to, from, aggregate });
+  const data = transformToFiles(jsonData);
+  drawChart(
+    data.files,
+    "top-files",
+    `Top 10 files downloaded from ${from.toLocaleDateString(
+      [],
+      dateTimeOptions
+    )}`
+  );
+  drawChart(
+    data.nodes,
+    "top-nodes",
+    `Top 10 nodes from ${from.toLocaleDateString([], dateTimeOptions)}`
+  );
+}
+
+anychart.onDocumentLoad(() => {
+  loadAndDraw();
+});
+
+const buttons = document.getElementsByClassName("period-button");
```

Comment:
You can use https://javascript.info/event-delegation or https://learn.javascript.ru/event-delegation and use https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes